logstash-core/spec/logstash/plugins/logging_spec.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/plugin"
+require "logstash/outputs/base"
+require "logstash/codecs/base"
+require "logstash/inputs/base"
+require "logstash/filters/base"
+require "logstash/logging"
+
+describe LogStash::Logging::Util::TopItems do
+
+  subject { described_class.new }
+
+  it "should be possible to add an item" do
+    subject.add("key", 1)
+    expect(subject.size).to eq(1)
+  end
+
+  it "should retrieve Top K in value order" do
+    subject.add("foo.bar.zet", 10)
+    subject.add("x.y.z", 5)
+    subject.add("a.b.c", 15)
+    expect(subject.top_k(2).map { |o| o.threshold }).to eq(["a.b.c", "foo.bar.zet"])
+  end
+
+  it "should keep only N elements in the set" do
+    11.times do |i|
+      subject.add("#{i}.foo.bar.#{i}", rand(20))
+    end
+    expect(subject.size).to eq(10)
+  end
```

Comment:
Consider expanding this test to verify the expected behaviour of the algorithm at the `SIZE` boundary.

When filled via `add("b", 100)` through `add("b", 109)` then `add("c", 200)` will have `"b", 109 -> 101` and `"c", 200` but `add("a", 50)` will displace `"c", 200`.

Its hard to understand why `"a", 50` is 'worth' more than `"c", 200`